aea/registries/base.py

```diff
@@ -570,9 +616,26 @@ def populate_skills(self, directory: str, agent_context: AgentContext) -> None:
                 "Processing the following skill directory: '{}".format(skill_directory)
             )
             try:
-                skill = Skill.from_dir(str(skill_directory), agent_context)
-                assert skill is not None
-                self.add_skill(skill)
+                skill_loader = ConfigLoader.from_configuration_type(
+                    ConfigurationType.SKILL
+                )
+                skill_config = skill_loader.load(
```

Comment:
basically we load the same configuration file twice: here and in `Skill.from_dir`. I tried to limit the changes to public methods. However, this part will be refactored when using the builder class.